gym_sts/envs/single_combat.py

```diff
@@ -0,0 +1,54 @@
+from gym_sts.spaces.observations import Observation
+
+from .base import SlayTheSpireGymEnv
+
+
+class SingleCombatSTSEnv(SlayTheSpireGymEnv):
+    def __init__(self, *args, enemy: str = "3_Sentries", **kwargs):
+        if "value_fn" not in kwargs:
+            kwargs["value_fn"] = single_combat_value
+        super().__init__(*args, **kwargs)
+        self.enemy = enemy
+
+    def reset(self, *args, **kwargs):
+        res = super().reset(*args, **kwargs)
+
+        obs = self.communicator.basemod(f"fight {self.enemy}")
+        assert obs.in_combat
+        self.observation_cache.append(obs)
+
+        if type(res) == tuple and len(res) == 2:
+            # params.return_info is set to true
+            info = {
+                "seed": self.seed,
+                "sts_seed": self.sts_seed,
+                "rng_state": self.prng.getstate(),
+                "observation": obs,
+            }
+            return obs.serialize(), info
+        else:
+            return obs.serialize()
+
+    def step(self, action_id: int):
+        ser, reward, should_reset, info = super().step(action_id)
+
+        if (
+            not info["observation"].in_combat
+            or info["observation"].screen_type != "NONE"
```

Comment:
This might work for the stock deck, but it might fail in cases where some sort of card selection needs to take place, i.e. a "GRID" or "HAND_SELECT" screen type.

Reply:
Not really suggesting we change this right now, but it's probably worth a TODO comment.

Reply:
I made it a bit simpler and added a comment